src/constraint_solver/constraint_solver.h

```diff
@@ -252,6 +252,7 @@ struct DefaultPhaseParameters {
 //
 class Solver {
  public:
+  // TODO: TBD(MWP): obsolete: not to oversimplify the issue, but I think that there is NO VALID REASON, in my estimation, for IntegerCastInfo, when we simply treat the AST nodes as what they are. Let the polymorphism virtual tables pick up what they are and that should be that.
```

Comment:
you are missing the point. But this issue is too complex, and is not a question of AST, nor virtual table.